src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1801,7 +1801,21 @@ private void VisitStatementsWithLocalFunctions(BoundBlock block)
             SyntaxNode syntax,
             bool isCall)
         {
-            if (Join(ref localFunctionState.StartingState, ref State) &&
+            // When we are in a conditional state after analyzing a call to a local function,
```

Comment:
>When we are in a conditional state after analyzing a call to a local function, [](start = 15, length = 78)

Consider adjusting the comment to reflect what state we are dealing with.

Reply:
It actually feels like we should be able to replace my change which temporarily merges the conditional states, with a `Debug.Assert(!IsConditionalState)` here. Since I do not expect to be in a conditional state after analyzing the call arguments, but before analyzing the call postconditions. Will verify.